test/python/test_dagcircuit.py

```diff
@@ -209,6 +209,34 @@ def test_nodes_in_topological_order(self):
         self.assertEqual([1, 3, 5, 7, 8, 9, 10, 11, 12, 13, 4, 14, 2, 15, 6],
                          [i for i in named_nodes])
 
+    def test_dag_has_edge(self):
+        """ Test that existence of edges between nodes is correctly identified"""
+        self.assertTrue(self.dag.has_edge(1, 2))
+        self.assertTrue(self.dag.has_edge(1, 2, (QuantumRegister(3, 'qr'), 0)))
+
+        self.assertFalse(self.dag.has_edge(1, 3))
+        self.assertFalse(self.dag.has_edge(1, 3, (QuantumRegister(3, 'qr'), 0)))
+
```

Comment:
Can you add a test case checking that we return false when an edge exists but doesn't match the specified wire?

Reply:
Sure, I have just added one